### PR TITLE
connect site with customer.io

### PR DESCRIPTION
**Pull request by mbouzi**
Connect Dev Waitlist Inputs on our site to segment in customer.io

**Form Locations**
- homepage hero
- homepage above the footer 
- /waitlist route

**Testing**
- error should appear if invalid email
- should redirect to thank you page upon submission
- user email should appear in customer.io in "dev_sign_up" segment (in testing environment)

Post approval, I'll move the email segment to the production environment before merging

### CODE REVIEW

**Comment by pgerbes1** [comment]
This lgtm

**Comment by pgerbes1** [comment]
@mbouzi can we add a description to this PR (and any going forward) describing what is being done and why. The engineering team has a good template they use for the Storj codebase. Really helps me when reviewing changes. Many thanks!

File: layouts/index.html [hunk 1]
```diff
@@ -14,3 +14,18 @@
...
   </body>
 </html>
+
+<script type="text/javascript">
+  window.onload = function() {
+    var button = document.getElementById("mc-embedded-subscribe")
+
+    button.onclick = function (event) {
+
+      var inputValue = document.getElementsByClassName("cta-email")[0].value
```
Comment (VinozzZ) [inline layouts/index.html]: we probably should check if the element exit before indexing into the array

File: layouts/waitlist/single.html [hunk 2]
```diff
@@ -23,4 +23,17 @@
...
   </body>
-</html>
\ No newline at end of file
+</html>
+<script type="text/javascript">
+  window.onload = function() {
+    var button = document.getElementById("mc-embedded-subscribe")
+
+    button.onclick = function (event) {
+      var inputValue = document.getElementById("mce-EMAIL").value
```
Comment (VinozzZ) [inline layouts/waitlist/single.html:32]: same as above. check that the element actually exit

File: layouts/index.html [hunk 3]
```diff
@@ -14,3 +14,21 @@
...
   </body>
 </html>
+
+<script type="text/javascript">
+  window.onload = function() {
+    var button = document.getElementById("mc-embedded-subscribe")
+
+    button.onclick = function (event) {
+
+      if(document.getElementsByClassName("cta-email")[0]) {
```
Comment (VinozzZ) [inline layouts/index.html]: Can you do something like this? 
```
var ctaEmail = document.getElementsByClassName("cta-email");
if (!ctaEmail || ctaEmail.length()) {
  return;
} 

var inputValue = ctaEmail.value
```

File: layouts/partials/waitlist/main.html [hunk 4]
```diff
@@ -56,10 +56,10 @@ <h2 class="waitlist__form-heading">
         <img class="img-fluid" src="{{ $marsImg.img_path }}" alt="{{ $marsImg.caption }}">
       </div>
     </div>
+    <div class="row mt-3 mb-5">
```
Comment (VinozzZ) [inline layouts/partials/waitlist/main.html]: Please remove this since this is only used for testing

File: layouts/partials/waitlist/main.html [hunk 5]
```diff
@@ -27,7 +27,7 @@ <h2 class="waitlist__form-heading">
             <p class="waitlist__form-description">
                 {{ replace .Params.formDescription "\n" "<br/>" | safeHTML}}
             </p>
-        </div> 
+        </div>
       <div id="mc_embed_signup">
         <form action="https://github.us8.list-manage.com/subscribe/post?u=edba9fc4ce8b9095a31859cba&amp;id=a5cc1e6ed3" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" class="validate" target="_blank" novalidate>
```
Comment (VinozzZ) [inline layouts/partials/waitlist/main.html]: We need to remove this form that's generated by mailchimp and replace the form with the logic we have to submit the email to customer.io

File: layouts/index.html [hunk 6]
```diff
@@ -17,18 +17,59 @@
...
     var button = document.getElementById("mc-embedded-subscribe")
 
     button.onclick = function (event) {
+      event.preventDefault()
 
-      if(document.getElementsByClassName("cta-email")[0]) {
+      if(document.getElementsByClassName("cta-email") && document.getElementsByClassName("cta-email")[0]) {
 
-        var inputValue = document.getElementsByClassName("cta-email")[0].value
+        var inputValue = document.getElementsByClassName("cta-email")[0].value;
```
Comment (VinozzZ) [inline layouts/index.html]: Why not move these two lines to be above the if statement and use the declared variables in the if statement instead.

File: layouts/index.html [hunk 7]
```diff
@@ -17,18 +17,59 @@
...
+      var buttomButton = document.getElementsByClassName("cta-btm-join")[0]
+
+      buttomButton.onclick = function (event) {
+        event.preventDefault()
+        if(document.getElementsByClassName("cta-btm-email") && document.getElementsByClassName("cta-btm-email")[0]) {
+
+          var input = document.getElementsByClassName("cta-btm-email")[0].value
+          var errElm = document.getElementById("mce-error-response-bottom");
+
+          if (!inputValue || inputValue.length < 0 || !inputValue.includes("@")) {
```
Comment (VinozzZ) [inline layouts/index.html:58]: I think here you meant to use `input` ?

**Pull request merged**